lib/pages/chat/chat_app_bar_title.dart

```diff
@@ -115,37 +95,51 @@ class ChatAppBarTitle extends StatelessWidget {
     );
   }
 
-  Widget _buildStatusContent(BuildContext context, Room room) {
+  StreamBuilder<ConnectivityResult> _buildStatusContent(BuildContext context, Room room) {
     final TextStyle? statusTextStyle = Theme.of(context).textTheme.labelSmall?.copyWith(
       fontSize: 13,
       color: Theme.of(context).colorScheme.tertiary,
       letterSpacing: ChatAppBarTitleStyle.letterSpacingStatusContent,
     );
 
-    if (room.getLocalizedTypingText(context).isEmpty) {
-      return Text(
-        room.getLocalizedStatus(context).capitalize(context),
-        maxLines: 1,
-        overflow: TextOverflow.ellipsis,
-        style: statusTextStyle,
-      );
-    } else {
-      return Row(
-        children: [
-          Text(
-            room.getLocalizedTypingText(context),
+    return StreamBuilder<ConnectivityResult>(
+      stream: controller.networkConnectionService.getStreamInstance(),
+      builder: (context, snapshot) {
```

Comment:
what happen if connection state not change, but the typing/online state change?

Reply:
After trying differents scenarios there is indeed a memory leak when disconnecting because typing ephemeral variable is not resynchronized. I'm working on it

Reply:
bug is unconsistent and seems related to spam disconnect / reconnect. I got no issue when user is disconnecting while someone is typing 
![disconnectduringtyping](https://github.com/linagora/twake-on-matrix/assets/48354990/f2506d8f-62bb-4e33-acf3-05495d58342a)

Typing when disconnected : 
![typingduringdisconnected](https://github.com/linagora/twake-on-matrix/assets/48354990/479809ce-306b-4224-97d0-87bfc8ad1e51)

Reply:
WHAT??? why we need to care about `typing` when disconnected?

Reply:
not only testing on the `disconnected`, please test on the `connected`

Reply:
please focus on the stream